test/com/google/javascript/jscomp/CommandLineRunnerTest.java

```diff
@@ -1459,6 +1458,123 @@ public void testProcessCJSWithClosureRequires() {
+                "goog.provide = function (arg) {};\n" +
+                "goog.require = function (arg) {};\n",
+
+            "goog.provide('goog.array');\n",
+
+            "goog.require('goog.array');\n" +
+                "function Baz() {}\n" +
+                "Baz.prototype = {\n" +
+                "  baz: function() {\n" +
+                "    return goog.array.last(['asdf','asd','baz']);\n" +
+                "  },\n" +
+                "  bar: function () {\n" +
+                "    return 4 + 4;\n" +
+                "  }\n" +
+                "}\n" +
+                "module.exports = Baz;",
+
+            "var Baz = require('./Baz');\n" +
+                "var baz = new Baz();\n" +
+                "console.log(baz.baz());\n" +
+                "console.log(baz.bar());\n"
+        },
+        new String[] {
+            "var goog=goog||{},COMPILED=!1;" +
+                "goog.provide=function(a){};goog.require=function(a){};",
+
+            "goog.array={};",
+
+            "function Baz$$module$Baz(){}" +
+                "Baz$$module$Baz.prototype={" +
+                "baz:function(){return goog.array.last([\"asdf\",\"asd\",\"baz\"])}," +
+                "bar:function(){return 8}" +
+                "};" +
+                "var module$Baz=Baz$$module$Baz;",
+
+            "var module$app={}," +
+                "Baz$$module$app=module$Baz," +
+                "baz$$module$app=new Baz$$module$app;" +
+                "console.log(baz$$module$app.baz());" +
+                "console.log(baz$$module$app.bar());"
+        });
+  }
+
+  public void testProcessCJSWithES6Export() {
+    args.add("--process_common_js_modules");
+    args.add("--entry_point=app");
+    args.add("--dependency_mode=STRICT");
+    args.add("--language_in=ECMASCRIPT6");
+    setFilename(0, "foo.js");
+    setFilename(1, "app.js");
+    test(new String[] {
+            "export default class Foo {"
+            + "  bar() { console.log('bar'); }"
+            + "}",
+
+            "var FooBar = require('./foo');\n" +
+                "var baz = new FooBar.default();\n" +
+                "console.log(baz.bar());"
+        },
+        new String[] {
+            "var module$foo={}," +
+                "Foo$$module$foo=function(){};" +
+                "Foo$$module$foo.prototype.bar=function(){console.log(\"bar\")};" +
+                "module$foo.default=Foo$$module$foo;",
+
+            "var module$app={}," +
+                "FooBar$$module$app=module$foo," +
+                "baz$$module$app=new FooBar$$module$app.default();" +
+                "console.log(baz$$module$app.bar());"
+        });
+  }
+
+  public void testES6ImportOfCJS() {
+    args.add("--process_common_js_modules");
+    args.add("--entry_point=app");
+    args.add("--dependency_mode=STRICT");
+    args.add("--language_in=ECMASCRIPT6");
+    setFilename(0, "foo.js");
+    setFilename(1, "app.js");
+    test(new String[] {
+            "/** @constructor */ function Foo () {}" +
```

Comment:
@MatrixFrog I was proposing we convert the styles of the tests as a separate PR as this one is pretty large already.